src/NovelAudioService.cpp

```diff
@@ -0,0 +1,173 @@
+//
+// Created by krjoh on 29/08/2019.
+//
+
+#include "NovelAudioService.h"
+#include <iostream>
+#include <functional>
+
+namespace NovelRT {
+
+NovelAudioService::NovelAudioService(){
+  initializeAudio();
+  _nextChannel = 0;
+}
+
+int NovelAudioService::initializeAudio(){
+  if (SDL_Init(SDL_INIT_AUDIO) < 0 )
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL Error: " << SDL_GetError() << std::endl;
+    isInitialized = false;
+    return -1;
+  }
+  if (Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT, 2, 2048) < 0)
+  {
+    std::cout << "ERROR: Cannot play audio!" << std::endl;
+    std::cout << "SDL_Mixer Error: " << Mix_GetError() << std::endl;
+    isInitialized = false;
+    return -1;
+  }
+  std::cout << "INFO: Allocating channels..." << std::endl;
+  if (Mix_AllocateChannels(2) < 0)
+  {
+    std::cout << "ERROR: Failed to allocate channels." << std::endl;
+    isInitialized = false;
+    return -1;
```

Comment:
can we change to `bool` please? :)

Reply:
Yep, we can. I was trying to maintain a likeness to how both SDL_mixer and SDL2 report errors, but I see that this can get confusing as hell for anyone looking for a failure. Will resolve when updated.